app/domain/authentication/authn/login.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'command_class'
+
+module Authentication
+  module Authn
+    Login = C= CommandClass.new(
+      dependencies: {
+        role_cls:        ::Role,
+        credentials_cls: ::Credentials
+      },
+      inputs:       [:authenticator_input]
+    ) do
+
+      extend Forwardable
+      def_delegators :@authenticator_input, :account, :credentials, :username
+
+      def call
+        return nil unless validate_role_has_credentials
+        authenticate
+        api_key
+      end
+
+      private
+
+      def validate_role_has_credentials
+        role_credentials
+      end
+
```

Comment:
this method is redundant. you can remove it and change the first line in the call function to `return nil unless role_credentials`

Reply:
this is not fixed as well.